php/class-connect.php

```diff
@@ -870,6 +919,92 @@ public function switch_account() {
 		return $return;
 	}
 
+	/**
+	 * Check the REST API connectivity for Cloudinary's endpoints.
+	 *
+	 * @return array
+	 */
+	public static function check_rest_api_connectivity() {
+
+		$connectivity = self::test_rest_api_connectivity();
+		$plugin       = get_plugin_instance();
+		$notices      = get_option( $plugin::KEYS['notices'], array() );
+		$now          = current_time( 'timestamp' ); // phpcs:ignore WordPress.DateTime.CurrentTimeTimestamp.Requested
+
+		wp_clear_scheduled_hook( 'cloudinary_rest_api_connectivity' );
+
+		if ( $connectivity['working'] ) {
+			unset( $notices[ self::META_KEYS['notices'] ] );
+			if ( empty( $notices ) ) {
+				delete_option( $plugin::KEYS['notices'] );
+			} else {
+				update_option( $plugin::KEYS['notices'], $notices, false );
+			}
+
+			wp_schedule_single_event( $now + ( DAY_IN_SECONDS ), 'cloudinary_rest_api_connectivity' );
+		} else {
+			update_option(
+				$plugin::KEYS['notices'],
+				array(
+					'rest_api_notices' => $connectivity,
+				),
+				false
+			);
+
+			wp_schedule_single_event( $now + ( HOUR_IN_SECONDS ), 'cloudinary_rest_api_connectivity' );
+		}
+
+		return $connectivity;
+	}
+
+	/**
+	 * Test the REST API connectivity for Cloudinary's endpoints.
+	 *
+	 * @return array
+	 */
+	public static function test_rest_api_connectivity() {
+		$result = array(
+			'working' => true,
+			'message' => __( 'Cloudinary was able to connect to the WordPress REST API.', 'cloudinary' ),
+		);
+
+		$timeout = 10;
+		$headers = array(
+			'Cache-Control' => 'no-cache',
+			'X-WP-Nonce'    => wp_create_nonce( 'wp_rest' ),
+		);
+
+		// This filter is documented in wp-includes/class-wp-http-streams.php.
+		$sslverify = apply_filters( 'https_local_ssl_verify', false );
+
+		$url      = rest_url( 'cloudinary/v1/test_rest_api' );
+		$response = wp_remote_get( $url, compact( 'headers', 'timeout', 'sslverify' ) );
```

Comment:
@pereirinha - perhaps use `wp_safe_remote_get` to keep linters happy.

and, any specific reason to use compact here over an array of args?